File: py/decrypt.py
# ==============================================================================
#
#    Use:
#    decrypt(b'WzM3NDQyLCA1MjQ4OCwgNTU4NzQsIDU2MDcyLCA1NzQxMCwgMTcxNjAsIDQ1MTIyLCA1NzYwOCwgNTg5NDYsIDU2MDcyLCA1MTc3OCwgMTc2NzJd', "key")
#    => "Hello World!"
#
# ==============================================================================

import base64

def encode(text):

        encoded_arr = [ord(i) for i in text]
        return encoded_arr

def decrypt(text, key):

	encoded_key = encode(key)
	decoded_b64 = eval(base64.b64decode(text))
	buf = []

	for x in decoded_b64:
		encoded_key = list(reversed(encoded_key))

		for i in encoded_key:
			x = x - 1 >> i % 8

		buf.append(x)

	decrypted_string = ''.join(chr(letter) for letter in buf)

	return decrypted_string

File: js/decrypt.js
/* ==========================================================================
 *
 *  Use:
 *  "WzM3NDQyLDUyNDg4LDU1ODc0LDU2MDcyLDU3NDEwLDE3MTYwLDQ1MTIyLDU3NjA4LDU4OTQ2LDU2MDcyLDUxNzc4XQ==".decrypt("key")
 *  => "Hello World!"
 *
 * ========================================================================== */

String.prototype.encode = function() {
    let array = [];
    for (let i of this) {
        array.push(i.charCodeAt(0))
    }
    return array
}
String.prototype.decrypt = function(key) {
    const keyEncoded = key.encode()
	if (typeof atob === 'undefined') {
        global.atob = b64Encoded => new Buffer(b64Encoded, 'base64').toString();
    }
    let array = JSON.parse(
        atob(this.toString())
    )
    let decrypted = array.map(x => {
		keyEncoded.reverse()
        x = parseInt(x)
        for (let i of keyEncoded) {
            x = x - 1 >> i % 8
        }
        return x;
    })
    return String.fromCharCode(...decrypted)
}

module.exports = (text, key) => text.decrypt(key)
